discord/widget.py

```diff
@@ -0,0 +1,217 @@
+
+    @property
+    def display_name(self):
+        """Returns the member's display name."""
+        return self.nick if self.nick else self.username
+
+    @property
+    def avatar_url(self):
+        """Returns the URL version of the member's avatar. Returns an empty string if it has no avatar."""
+        return self.avatar_url_as()
+
+    def avatar_url_as(self, *, format='webp', size=1024):
+        """:class:`str`: The same operation as :meth:`User.avatar_url_as`."""
+        if not valid_icon_size(size):
+            raise InvalidArgument("size must be a power of 2 between 16 and 4096")
+        if format not in VALID_ICON_FORMATS:
+            raise InvalidArgument("format must be one of {}".format(VALID_ICON_FORMATS))
+
+        if self.avatar is None:
+            return self.default_avatar_url
+
+        return 'https://cdn.discordapp.com/avatars/{0.id}/{0.avatar}.{1}?size={2}'.format(self, format, size)
+
+    @property
+    def default_avatar(self):
+        """:class:`DefaultAvatar`: The same operation as :meth:`User.default_avatar`."""
+        return DefaultAvatar(int(self.discriminator) % len(DefaultAvatar))
+
+    @property
+    def default_avatar_url(self):
+        """Returns a URL for a user's default avatar."""
+        return 'https://cdn.discordapp.com/embed/avatars/{}.png'.format(self.default_avatar.value)
+
+class Widget:
+    """Represents a :class:`Guild` widget.
+
+    This model is always given, as long as it is enabled.
+
+    Attributes
+    -----------
+    id: :class:`int`
+        The guild's ID.
+    name: :class:`str`
+        The guild's name.
+    channels: List[:class:`VoiceChannel`]
+        The accessible voice channels in the guild.
+    members: List[:class:`Member`]
+        The online members in the server. Note: offline
+        members do not appear in the widget.
+    invite: Optional[:class:`Invite`]
+        The invite set for the widget.
+    """
+    __slots__ = ('_state', 'channels', 'invite', 'id', 'members', 'name')
+
+    def __init__(self, *, data, invite):
+        self.invite = invite
+        self.name = data.get('name')
+        self.id = int(data.get('id'))
+
+        self.channels = []
+        for channel in data.get('channels'):
+            _id = int(channel['id'])
+            self.channels.append(WidgetChannel(id=_id, name=channel['name'], position=channel['position']))
+
+        self.members = []
+        for member in data.get('members'):
+            _id = int(member.get('id'))
+            bot = bool(member.get('bot'))
+            status = try_enum(Status, member.get('status'))
+            deafened = bool(member.get('deaf')) or bool(member.get('self_deaf'))
+            muted = bool(member.get('mute')) or bool(member.get('self_mute'))
+            suppress = bool(member.get('mute'))
+
+            game = None
+            if member.get('game'):
+                game = Game(**member.get('game'))
+
+            connected_channel = None
```

Comment:
```suggestion
            connected_channel = utils._get_as_snowflake(member, 'channel_id')
```

Followed by `if connected_channel:`